### PR TITLE
Feature/redid card

### PR DESCRIPTION
**Pull request by azabolotnaiaGD**
Redid cart ( one cart instead of many) , changed mapping , deleted ApiResponse

### CODE REVIEW

**Review by acostasgd — APPROVED** [review]

File: src/main/java/com/nvs/store/controllers/CartController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/carts")
+public class CartController {
+
+    private final CartService cartService;
+
+
+    @PostMapping()
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/controllers/CartController.java:19]: please remove brackets if not needed

File: src/main/java/com/nvs/store/dto/cart/CartItem.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package com.nvs.store.dto.cart;
+
+import jakarta.persistence.*;
+
+@Entity
+@Table(name = "cart_items")
+public class CartItem {
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/dto/cart/CartItem.java:7]: please override equals/hashcode/toString methods to correct functionality

File: src/main/java/com/nvs/store/dto/cart/ProductDto.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.nvs.store.dto.cart;
+
+import javax.validation.constraints.NotNull;
+
+public class ProductDto {
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/dto/cart/ProductDto.java:5]: ditto - need equals/hashcode/toString override

File: src/main/java/com/nvs/store/models/cart/Cart.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package com.nvs.store.models.cart;
+
+import com.nvs.store.dto.cart.CartItem;
+import jakarta.persistence.*;
+import java.util.Date;
+import java.util.List;
+
+@Entity
+@Table(name = "carts")
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/models/cart/Cart.java:9]: same here

Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/models/cart/Cart.java:9]: same here

File: src/main/java/com/nvs/store/repository/CartItemRepository.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.nvs.store.repository;
+
+import com.nvs.store.dto.cart.CartItem;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface CartItemRepository extends JpaRepository<CartItem, Long> {
+//    ItemId findByProductId(Long id);
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/repository/CartItemRepository.java:9]: remove comment if not used

File: src/main/java/com/nvs/store/repository/ProductRepository.java [hunk 6]
```diff
@@ -8,6 +8,8 @@
 public interface ProductRepository extends JpaRepository<Product, Long> {
     Product getProductById(Long id);
 
-    Product  findTopByOrderByIdDesc();
+    Product findTopByOrderByIdDesc();
+
+    List<Product> findAll();
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/repository/ProductRepository.java:13]: I guess you dont need to specify this method explicitely - its part of jpa repository

File: src/main/java/com/nvs/store/service/CartService.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            totalCost = totalCost.add(price.multiply(BigDecimal.valueOf(cartItem.getQuantity())));
+        }
+        CartDto cartDto = new CartDto();
+        cartDto.setCartItems(cartItemsList);
+        cartDto.setTotalCost(totalCost);
+        return cartDto;
+    }
+
+    public CartDto updateCartItems(Long productId, Integer quantity) {
+        CartItem cartItem = cartItemRepository.findById(productId)
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/service/CartService.java:51]: I think you should pass here cart ID - not product id

File: src/main/java/com/nvs/store/service/CartService.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+    public CartDto addToCart(ProductDto productDto) {
+        Product product = productService.findById(productDto.getProductId());
+        if (productDto.getQuantity() > product.getAvailable()) {
+            throw new NotAvailableException("We don't have enough available quantity of this product. We have only:" + product.getAvailable());
+        }
+        CartItem cartItem = new CartItem();
+        cartItem.setQuantity(productDto.getQuantity());
+
+        cartItemRepository.save(cartItem);
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/service/CartService.java:33]: to which cart will be added item?

File: src/main/resources/application.yml [hunk 9]
```diff
@@ -6,7 +6,7 @@ spring:
     driver-class-name: com.mysql.cj.jdbc.Driver
   jpa:
     hibernate:
-      ddl-auto: update
+      ddl-auto: create-drop
```
Comment (ppanasencoGD) [inline src/main/resources/application.yml:9]: revert

**Pull request closed without merging**